runelite-client/src/main/java/net/runelite/client/plugins/pvptools/PvpToolsPlugin.java

```diff
@@ -422,6 +425,14 @@ private void onPlayerDespawned(PlayerDespawned event)
 		}
 	}
 
+	private void onMenuEntryAdded(MenuEntryAdded event)
+	{
+		if (config.hideCastNpcs() && event.getType() == MenuAction.SPELL_CAST_ON_NPC.getId())
```

Comment:
We've got a convention-ish thing about caching config values in the plugin until they change (reactive vs iterative anyone?). You should add that to onConfigChanged